pkg/controller/machinedeployment/controller.go

```diff
@@ -229,10 +229,11 @@ func (r *ReconcileMachineDeployment) reconcile(ctx context.Context, d *v1alpha1.
 	}
 
 	// Add foregroundDeletion finalizer if MachineDeployment isn't deleted and linked to a cluster.
-	if cluster != nil && d.ObjectMeta.DeletionTimestamp.IsZero() {
-		if !util.Contains(d.Finalizers, metav1.FinalizerDeleteDependents) {
-			d.Finalizers = append(d.ObjectMeta.Finalizers, metav1.FinalizerDeleteDependents)
-		}
+	if cluster != nil &&
+		d.ObjectMeta.DeletionTimestamp.IsZero() &&
+		!util.Contains(d.Finalizers, metav1.FinalizerDeleteDependents) {
```

Comment:
Wouldn't this cause Update to be skipped if the finalizer is already present?

Reply:
actually, after reviewing past the diff context, I don't think this is an issue.